src/EventHandler.ts

```diff
@@ -1,304 +1,346 @@
+      // This was a single tap on a link. Do nothing.
+      this.pendingTouchEventEnd = null;
+      return;
     }
 
-    private handleLinks = (event: MouseEvent | TouchEvent): void => {
-        const link = this.checkForLink(event);
-        if (link) {
-            // Open external links in new tabs.
-            const isSameOrigin = (
-                window.location.protocol === link.protocol &&
-                window.location.port === link.port &&
-                window.location.hostname === link.hostname
-            );
-            const isInternal = (link.href.indexOf("#"));
-            if (!isSameOrigin) {
-                window.open(link.href, "_blank");
-                event.preventDefault();
-                event.stopPropagation();
-            } else if (isSameOrigin && isInternal !== -1) {
-                this.onInternalLink(event);
-            } else if (isSameOrigin && isInternal === -1) {
-                link.click();
-            }
-        }
+    // This was a single tap.
+    const touch = this.pendingTouchEventEnd.changedTouches[0];
+    if (!touch) {
+      // Somehow we got a touch event with no touches.
+      return;
     }
 
-    private handleKeyboard = (event: KeyboardEvent): void => {
-        const LEFT_ARROW = 37;
-        const RIGHT_ARROW = 39;
-        const TAB_KEY = 9;
-
-        if (event.keyCode === LEFT_ARROW) {
-            this.onLeftArrow(event);
-        } else if (event.keyCode === RIGHT_ARROW) {
-            this.onRightArrow(event);
-        } else if (event.keyCode === TAB_KEY) {
-            event.preventDefault();
+    const x = touch.clientX;
+    const width = window.innerWidth;
+    if (x / width < 0.3) {
+      this.onLeftTap(this.pendingTouchEventEnd);
+    } else if (x / width > 0.7) {
+      this.onRightTap(this.pendingTouchEventEnd);
+    } else {
+      this.onMiddleTap(this.pendingTouchEventEnd);
+    }
+    this.pendingTouchEventEnd = null;
+    return;
+  };
+
+  private checkForLink = (
+    event: MouseEvent | TouchEvent
+  ): HTMLAnchorElement | null => {
+    let nextElement = event.target as Element;
+    while (nextElement && nextElement.tagName.toLowerCase() !== "body") {
+      if (
+        nextElement.tagName.toLowerCase() === "a" &&
+        (nextElement as HTMLAnchorElement).href
+      ) {
+        return nextElement as HTMLAnchorElement;
+      } else {
+        (nextElement as any) = nextElement.parentElement;
+      }
+    }
+    return null;
+  };
+
+  // private handleMouseMove = (event: MouseEvent): void => {
+  //     const x = event.clientX;
+  //     const width = window.innerWidth;
+  //     if (x / width < 0.3) {
+  //         this.onLeftHover();
+  //     } else if (x / width > 0.7) {
+  //         this.onRightHover();
+  //     } else {
+  //         this.onRemoveHover();
+  //     }
+  // }
+
+  private handleMouseLeave = (): void => {
+    this.onRemoveHover();
+  };
+
+  private handleLinks = (event: MouseEvent | TouchEvent): void => {
```

Comment:
The new thing in this file is the addition of `isOnEpubHost` here.  I will tag the same thing in the original diff to help compare.  I'm also gonna file a "enforce a TS formatter" ticket because this is driving me nuts.